NEWS.md

```diff
@@ -1,5 +1,14 @@
 # rsconnect 0.8.30 (development version)
 
+* `deployApp()` now derives `appName` from `appDir` and `appPrimaryDoc`, 
+  never using the title. It also tells you what name has used (#538).
+  It now only simplifies the path if you are publishing to shinyapps.io,
+  since its restrictions on application names are much tighter than those
+  of Posit Connect.
+  
+* `generateAppName()` has been deprecated as using the `appTitle` to generate
```

Comment:
How does this change alter the workflow that folks use to repeatedly deploy the same content?

Today, I can use the publishing drop-down "Other Destination..." to repeatedly deploy the same Shiny application to a single server. They happen to have the same title (because I didn't override the IDE-suggested default) and come from the same directory (`shiny`, in this case).

The content I just created were given names "shiny3" and "shiny4".

Reply:
Hmmm, good question. Looks like the IDE is using `generateAppName()` for that, https://github.com/rstudio/rstudio/blob/3fba3707760ee74e14f1d682ec49e726084ac3d3/src/cpp/session/modules/SessionRSConnect.R#L592, so I'll undeprecate.

Reply:
I also realised that we should be using `generateAppName()` in https://github.com/rstudio/rsconnect/blob/defaultAppName/R/deployApp.R#L546-L549 to be consistent with the IDE, but that will take some refactoring in order to get the updated `appName` in the right place.

I've undeprecated `generateAppName()`